### DISCUSSION TITLE
how to write dynamic parameterized query in prisma?

### DISCUSSION THREAD

**Discussion by SalmanGits** [prisma/prisma#22571] [Q&A]
### Question

so i am trying 
```ts
 let base_query = `public."Ticket" as ticket join public."User" as u on u._id = ticket.user`;

      
        let dynamicQuery = '';
        const queryParams = [];

        if (req.body.status) {
            dynamicQuery += ` AND status = $${queryParams.length + 1}`;
            queryParams.push(req.body.status);
        }

        if (req.body.id) {
            dynamicQuery += ` AND user = $${queryParams.length + 1}`;
            queryParams.push(req.body.id);
        }

        const fullQuery = `SELECT * FROM ${base_query} WHERE 1=1${dynamicQuery}`;
      

        const tickets = await prisma.$queryRaw(Prisma.sql(fullQuery,...queryParams));
        
   ```     
        this does not seem to work

### Information about Prisma Schema, Client Queries and Environment (optional)

TypeError: Expected 96 strings to have 95 values
    
    
    
    i know it is somehting about my params only but i dont know what

**Comment by nurul3101** [comment]
Hello @SalmanGits 👋 

Based on the error message, it looks like there is a mismatch between the placeholders in your SQL query and the number of parameters you are passing to `Prisma.sql`.

Can you check if the number of placeholders in your query matches the number of elements in the `queryParams` array?

**Reply by SalmanGits** [reply]
no but i think we should improve the raw query support like typeorm

**Reply by ludralph** [reply]
Hi @SalmanGits 

Thanks for the suggestion. Can you please create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=)  and document the improvements you would like to see?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
